rasa/shared/utils/validation.py

```diff
@@ -241,24 +241,25 @@ def validate_training_data_format_version(
         )
         return True
 
+    if str(version_value) < LATEST_TRAINING_DATA_FORMAT_VERSION:
```

Comment:
I think this should actually come after L260, just before L261 `if`, to prioritise raising the `TypeError` if a legacy version is being.